### PR TITLE
CSP-710 Edit Your Profile Area of Interest

### PR DESCRIPTION
**Pull request by ManishAgarwalDfE**

### CODE REVIEW

File: src/SFA.DAS.Aan.SharedUi.UnitTests/Models/AmbassadorProfileViewModelsTests/PersonalDetailsViewModelTests.cs [hunk 1]
```diff
@@ -32,7 +33,7 @@ public void Setup()
         memberProfiles.ToArray()[1].PreferenceId = memberPreferences.ToArray()[1].PreferenceId;
         memberPreferences.ToArray()[1].Value = false;
         userType = MemberUserType.Apprentice;
-        var personalDetails = new PersonalDetailsModel(fullName, regionName, userType, personalDetailsChangeUrl);
+        var personalDetails = new PersonalDetailsModel(fullName, regionName, userType, personalDetailsChangeUrl, areaOfInterestChangeUrl);
         sut = new PersonalDetailsViewModel(personalDetails, memberProfiles, memberPreferences);
     }
```
Comment (MarkFCain) [inline src/SFA.DAS.Aan.SharedUi.UnitTests/Models/AmbassadorProfileViewModelsTests/PersonalDetailsViewModelTests.cs:38]: add a test to sut.areaOfInterest is the url you've set

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/EditAreaOfInterestControllerTests/EditAreaOfInterestControllerGetTests.cs [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+using SFA.DAS.ApprenticePortal.Authentication.TestHelpers;
+using SFA.DAS.Testing.AutoFixture;
+
+namespace SFA.DAS.ApprenticeAan.Web.UnitTests.Controllers;
+
+public class EditAreaOfInterestControllerGetTests
+{
+    static readonly string YourAmbassadorProfileUrl = Guid.NewGuid().ToString();
+
+    [Test]
```
Comment (MarkFCain) [inline src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/EditAreaOfInterestControllerTests/EditAreaOfInterestControllerGetTests.cs:24]: these 2 lines can be combined as [Test, MoqAutoData],  Same applies to all instances in this file

**Review by MarkFCain — APPROVED** [review]

File: src/SFA.DAS.ApprenticeAan.Web/Controllers/NetworkDirectoryController.cs [hunk 3]
```diff
@@ -28,6 +28,10 @@ public NetworkDirectoryController(IOuterApiClient outerApiClient)
     [HttpGet]
     public async Task<IActionResult> Index(NetworkDirectoryRequestModel request, CancellationToken cancellationToken)
     {
+        if (request.PageSize == null)
+        {
+            request.PageSize = 10;
```
[the remark this answers is not in the document]
Comment (ManishAgarwalDfE) [reply src/SFA.DAS.ApprenticeAan.Web/Controllers/NetworkDirectoryController.cs:33]: It inadvertently was pushed in while I was looking into pagination problem raised by Sudhakar earlier. Apologies.

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/EditAreaOfInterestControllerTests/EditAreaOfInterestControllerGetTests.cs (hunk 2 above)
Comment (ManishAgarwalDfE) [reply src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/EditAreaOfInterestControllerTests/EditAreaOfInterestControllerGetTests.cs]: Done

**Pull request merged**